### PR TITLE
models integration

### LINKED ISSUE #2: Implement getting data from Scorecard API
Subtasks:

- [x] Make sure different type of errors are handled and relevant exceptions raised
- [x]  Data should be stored in objects
- [ ] Research rate-limiting on API present or not
- [x] Test this for Github (a couple different ecosystems which we can scan later too) and Gitlab projects

### LINKED ISSUE #3: Create models for storing Scorecard Data
This is:
- [x] Abstract models to store data for a score and all it's checks
- [x] Django mixins to be used in SCIO and purldb
- [x] Help texts with these

Nice to have:
- [ ] More data about the types of checks, their relevance

### PR DESCRIPTION
**Pull request by 404-geek**
This PR is meant for the overall development of scorecode which includes

  - Fetching data from OSSF scorecard API
  - Storing them in objects
  - Saving them into your specfic data models if required.

Fixes #3 
Fixes #2

### CODE REVIEW

**Comment by AyanSinhaMahapatra** [comment]
Are all the comments at https://github.com/nexB/ScoreCode/pull/4#pullrequestreview-2151001759 and https://github.com/nexB/ScoreCode/pull/1#pullrequestreview-2132216329 resolved?

Please reply to them and resolve/close those individually, otherwise hard to discuss and track feedback already given.

We also need proper tests running in a CI before we can merge this too.

File: README.rst [hunk 1]
```diff
@@ -1,3 +1,55 @@
...
+
+    from scorecard import GetScorecard
+
+    # Define platform, organization, and repository
+    platform = "github.com"
+    org = "nexB"
+    repo = "scancode-toolkit"
+
+    # Fetch the scorecard
+    data = GetScorecard(platform, org, repo)
```
Comment (AyanSinhaMahapatra) [inline README.rst]:
```suggestion
    data = GetScorecard(platform="github.com", org="nexB", repo="scancode-toolkit")
```
Have this together.

File: pyproject.toml [hunk 2]
```diff
@@ -7,6 +7,78 @@ build-backend = "setuptools.build_meta"
 # and when there is .git dir and/or there is no git installed
 fallback_version = "9999.$Format:%h-%cs$"
 
+
+[project]
+name = "ScoreCode"
```
Comment (AyanSinhaMahapatra) [inline pyproject.toml]: We already have this info as a manifest in `setup.cfg`, we should have one or the another, not both.

File: setup.cfg [hunk 3]
```diff
@@ -54,18 +54,18 @@ docs =
...
+lint =
+    isort
+    black
+    mypy
+test =
+    pytest
+build =
+    setuptools
+    wheel
+sqlalchemy =
```
Comment (AyanSinhaMahapatra) [inline setup.cfg]: What is `sqlalchemy` used for?

**Review by AyanSinhaMahapatra — CHANGES_REQUESTED** [review]
@404-geek Thanks for the changes, looking much better, we are getting there.

File: pyproject.toml (hunk 2 above)
Comment (404-geek) [reply pyproject.toml]: I also thought the same but for some reason while i run `make dev` it is taking all dependencies from `pyproject.toml`

File: setup.cfg (hunk 3 above)
Comment (404-geek) [reply setup.cfg]: removed, sorry didn't notice it

File: README.rst (hunk 1 above)
Comment (404-geek) [reply README.rst]: Done

File: pyproject.toml (hunk 2 above)
Comment (AyanSinhaMahapatra) [reply pyproject.toml]: Have you tried removing the `pyproject.toml` and only having the `setup.cfg` like in the skeleton?

Comment (404-geek) [reply pyproject.toml]: have removed it [link](https://github.com/aboutcode-org/scorecode/pull/5/commits/63b335ddc3940e412e60363e506866305ceaf276#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711)

File: tests/test_skeleton_codestyle.py [hunk 4]
```diff
@@ -7,9 +7,9 @@
 # See https://aboutcode.org for more information about nexB OSS projects.
```
Comment (AyanSinhaMahapatra) [inline tests/test_skeleton_codestyle.py]: Can you remove this file? This is not required.

File: README.rst [hunk 5]
```diff
@@ -1,3 +1,55 @@
...
+- Supports various platforms and repositories
+
+Installation
+------------
+
+To install Scorecard, you can use pip:
+
+.. code-block:: bash
+
+    pip install scorecard
```
Comment (AyanSinhaMahapatra) [inline README.rst]: This is taken we have to choose something else:
https://pypi.org/project/scorecard/

Maybe `ossfscorecard`?

Comment (AyanSinhaMahapatra) [reply README.rst]: We also have to modify the package name in setup.cfg accordingly

File: setup.cfg [hunk 6]
```diff
@@ -23,28 +21,32 @@ classifiers =
...
 license_files =
     LICENSE
     - AUTHORS.rst
     - README.rst
     - CONTRIBUTING.rst
     - CHANGELOG.rst
+    - apache-2.0.LICENSE
 
 [options]
-python_requires = >=3.8
```
Comment (AyanSinhaMahapatra) [inline setup.cfg:38]: We should keep these options as I think they are needed to create wheels/source archives for a release

**Review by AyanSinhaMahapatra — CHANGES_REQUESTED** [review]
There are also some tests failing in azure:
You should modify https://github.com/aboutcode-org/scorecode/blob/main/azure-pipelines.yml#L14 to only run tests from the `tests` directory. So it would be `venv/bin/pytest -n 2 -vvs tests`

Also https://github.com/aboutcode-org/scorecode/blob/main/src/README.rst can be removed.

File: src/ossf_scorecard/contrib/models.py [hunk 7]
```diff
@@ -0,0 +1,160 @@
+import attr
```
Comment (AyanSinhaMahapatra) [inline src/ossf_scorecard/contrib/models.py]: This should not be in `src/ossf_scorecard/contrib/models.py`, rather `src/ossf_scorecard/models.py`?

File: README.rst (hunk 5 above)
Comment (AyanSinhaMahapatra) [reply README.rst]: Sorry, this would just be `scorecode` which is there in setup.cfg

**Comment by AyanSinhaMahapatra** [comment]
Let's also make sure all the code files have a notice and let's use this one everywhere: https://github.com/aboutcode-org/scorecode/blob/main/src/ossf_scorecard/__init__.py#L2

**Review by AyanSinhaMahapatra — APPROVED** [review]
LGTM! Thanks++ @404-geek 
Looking much better now.
Let's push a tag and release after merging! :rocket:

**Pull request merged**